CONTRIBUTING.md

```diff
@@ -1,22 +1,155 @@
 
+## <a name="coc"></a> Code of Conduct
+
+Help us keep Cartesi an open space for development.
+Please read and follow our [Code of Conduct](https://github.com/cartesi/rollups/blob/main/CODE_OF_CONDUCT.md).
+
+## <a name="basic"></a> Basic Contributing Guidelines
+
+We use the same guidelines for contributing code to any of our repositories, any developers wanting to contribute to Cartesi must create pull requests.
+This process is described in the [GitHub documentation](https://help.github.com/en/articles/creating-a-pull-request).
+Each pull request should be started against the main branch in the respective Cartesi repository.
+After a pull request is submitted the Cartesi team will review the submission and give feedback via the comments section of the pull request.
+After the submission is reviewed and approved, it will be merged into the master branch of the source.
 Please note the below! We appreciate everyone following the guidelines.
 
 * No --force pushes or modifying the Git history in any way;
 * Use non-master branches, using a short meaningful description, with words separated by dash (e.g. 'fix-this-bug');
 * All modifications must be made in a pull-request to solicit feedback from other contributors.
 
-## Get in Touch
+## <a name="commit"></a> Commit Message Format
+
+We have very precise rules over how our Git commit messages must be formatted, following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) guideline.
+This format leads to **easier to read commit history**.
+
+Each commit message consists of a **header**, a **body**, and a **footer**.
+
+```
+<header>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The `header` is mandatory and must conform to the [Commit Message Header](#commit-header) format.
+
+The `body` is optional.
+When the body is present it must be at least 20 characters long and must conform to the [Commit Message Body](#commit-body) format.
+
+The `footer` is optional.
+The [Commit Message Footer](#commit-footer) format describes what the footer is used for and the structure it must have.
+
+
+### <a name="commit-header"></a>Commit Message Header
+
+```
+<type>(<scope>): <subject>
+  │       │          │
+  │       │          └─⫸ Summary in present tense. Not capitalized. No period at the end.
+  │       │
+  │       └─⫸ Commit Scope: Check the available scopes of the repository.
+  │
+  └─⫸ Commit Type: build|feat|fix|test|perf|refactor|docs|style
+```
+
+The `<type>` and `<summary>` fields are mandatory, the `(<scope>)` field is optional.
```

Comment:
Adding to @gligneul's comment, we should say it here that the `!` is optional too.

Reply:
Okay.